kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/InCommitTimestampUtils.java

```diff
@@ -75,4 +77,20 @@ private static boolean didCurrentTransactionEnableICT(
             && IN_COMMIT_TIMESTAMPS_ENABLED.fromMetadata(readSnapshot.getMetadata());
     return isICTCurrentlyEnabled && !wasICTEnabledInReadSnapshot;
   }
+
+  /**
+   * Converts the given {@link ColumnVector} containing commit info to a {@link CommitInfo} object.
+   *
+   * @param commitInfoVector a ColumnVector containing commit info
+   * @return the commit info as a {@link CommitInfo} object, or an empty Optional if no commit info
+   *     is found
+   */
+  public static Optional<CommitInfo> getCommitInfo(ColumnVector commitInfoVector) {
```

Comment:
Why not move this to the CommitInfo class?

Reply:
+1, putting CommitInfo in ICT utils is confusing.